snuba/query/dsl.py

```diff
@@ -10,6 +10,13 @@ def literals_tuple(alias: Optional[str], literals: Sequence[Literal]) -> Functio
     return FunctionCall(alias, "tuple", tuple(literals))
 
 
+# Array functions
+def array_element(
```

Comment:
you sure about this?  Assuming we fix the visitor, this would probably be the only place in the code base where we do camel case.
Anyway, very weak opinion, I don't think either way ruins readability.

Reply:
I thought I responded to this but apparently I forgot. I don't have a strong opinion about this, but I think it's OK for the DSL to bend some of the rules if it helps it maintain consistency with the domain it's supposed to represent.

Another example is I think operator overloading is almost always a bad idea, but it seems like it might make sense to replace (or supplement) things like the arithmetic operators with operators on `Expression` that return a new `Expression` with the operator function applied.